lib/money/money_parser.rb

```diff
@@ -82,9 +82,12 @@ def extract_amount_from_string(input, currency, strict)
     number = number.to_s.strip
 
     if number.empty?
-      raise MoneyFormatError, "invalid money string: #{input}" if strict
-      Money.deprecate("invalid money strings will raise in the next major release \"#{input}\"")
-      return '0'
+      if Money.config.legacy_deprecations && !strict
+        Money.deprecate("invalid money strings will raise in the next major release \"#{input}\"")
+        '0'
```

Comment:
Does this still need to be `return '0'`?

Reply:
Sorry - I meant, this line is just `'0'` - should it be `return '0'`? Otherwise this '0' is just immediately thrown out?

Reply:
oh sorry! Will open a new PR to fix this